### PR TITLE
Feature/jdbc + Feature/grep

### PR DESCRIPTION
**Pull request by chopgye**

### CODE REVIEW

**Comment by chopgye** [comment]
I closed my PR for the grep app and just included it in this PR. Feature/jdbc is the most up to date branch.

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+    try {
+      String line;
+      BufferedReader reader = new BufferedReader(new FileReader(inputFile));
+      while ((line = reader.readLine()) != null) {
+        lines.add(inputFile + ":" + line);
+      }
+      reader.close();
+    } catch (FileNotFoundException ex) {
+      this.logger.error(ex.getMessage(), ex);
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java:111]: this.logger.error("App specific and meaningful message", ex);

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    for (String line : lines) {
+      bufferedWriter.write(line);
+      bufferedWriter.newLine();
+    }
+    bufferedWriter.close();
+  }
+
+  public static void main(String[] args) {
+    if (args.length != 3) {
+      throw new IllegalArgumentException("");
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java:138]: please put a message

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+    JavaGrepImp javaGrepImp = new JavaGrepImp();
+    javaGrepImp.setRegex(args[0]);
+    javaGrepImp.setRootPath(args[1]);
+    javaGrepImp.setOutFile(args[2]);
+
+    try {
+      javaGrepImp.process();
+    } catch (IOException ex) {
+      javaGrepImp.logger.error(ex.getMessage(), ex);
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java:152]: replace `ex.getMessage()` with a meaningful and app specific message

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    BasicConfigurator.configure();
+
+    JavaGrepImp javaGrepImp = new JavaGrepImp();
+    javaGrepImp.setRegex(args[0]);
+    javaGrepImp.setRootPath(args[1]);
+    javaGrepImp.setOutFile(args[2]);
+
+    try {
+      javaGrepImp.process();
+    } catch (IOException ex) {
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java:151]: you want to catch all exception `Exception ex`

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  public List<File> listFiles(String rootDir) {
+    List<File> files = new ArrayList<>();
+    File dir = new File(rootDir);
+    try {
+      Files.walk(dir.toPath())
+          .filter(Files::isRegularFile)
+          .forEach(path -> files.add(path.toFile()));
+
+    } catch (IOException ex) {
+      logger.error(ex.getMessage(), ex);
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java:43]: fix this too

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+    } catch (IOException ex) {
+      this.logger.error(ex.getMessage(), ex);
+    }
+    return null;
+  }
+
+  public static void main(String[] args) {
+    if (args.length != 3) {
+      throw new IllegalArgumentException("");
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java:66]: fix all exception occurarnces

File: core_java/grep/src/main/java/ca/jrvs/apps/practice/HelloWorld.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package ca.jrvs.apps.practice;
+
+class HelloWorld {
+
+  // Your program begins with a call to main().
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/practice/HelloWorld.java:5]: remove this file

**Pull request closed without merging**